ThunderTable/ScrollOffsetManagable.swift

```diff
@@ -0,0 +1,35 @@
+//
+//  ScrollDelegatable.swift
+//  ThunderTable
+//
+//  Created by Simon Mitchell on 02/09/2020.
+//  Copyright © 2020 threesidedcube. All rights reserved.
+//
+
+import UIKit
+
+/// A simpler version of `UIScrollViewDelegate` so we don't intefere
+/// with users `UIScrollViewDelegate` implementations
+public protocol ScrollOffsetDelegate: class {
+    
+    /// Called when the content offset of the scroll view changes due to `scrollViewDidScroll`
+    /// - Parameters:
+    ///   - scrollable: The scrollable that the change was for
+    func scrollViewDidChangeContentOffset(_ scrollable: ScrollOffsetManagable)
+}
+
+/// A protocol implemented to allow control and listening to scroll view offset changes
+/// by `ScrollOfffsetManager`
+public protocol ScrollOffsetManagable: class {
+    
+    /// The scroll view that is controllable on the object
+    var scrollView: UIScrollView? { get }
+    
+    /// The delegate to have scroll view delegate methods passed to, this should be a `weak`
+    /// property to avoid retain cycles.
+    /// - Warning: It is your job to call the method on this delegate from your scrollViewDidScroll method.
+    var scrollDelegate: ScrollOffsetDelegate? { get set }
+    
+    /// An identifier used by `ScrollOffsetManager`
+    var identifier: AnyHashable? { get set }
```

Comment:
I can defo see why if the implementor providing an identifier is real nice. Hence this. But it's a little annoying to have to provide it. Maybe I'm overlooking here, how's this going to work if some provide it and some don't?

Anyway we probably want some default implementations here? :) Like this and delegate as `nil`?

Reply:
So `TableViewController` actually sets this to the given `IndexPath` so all implementers have to do is provide the property. Not sure if that makes sense to you? It just means all information is stored on the cell itself, and we don't need to fetch different bits of info from different places. Maybe I need to make that clearer in the docs?

Reply:
Well Apple choose not to put an `IndexPath` property on the `UITableViewCell`. That's where the manager would hold onto that mapping. I'll come back to